kano_updater/scenarios.py

```diff
@@ -908,22 +908,53 @@ def beta_370_to_beta_380(self, dummy_progress):
 
         self._bootconfig_set_value_helper("gpu_mem", "256")
 
-        new_apps = [
-            'adventure',
-            'openttd',
-            'tux-paint',
-            'tux-typing',
-            'libreoffice',
-            'numpty-physics',
-            'gmail',
-            'google-drive',
-            'google-maps',
-            'wikipedia',
-            'whatsapp'
-        ]
-
-        for app in new_apps:
-            run_cmd_log('kano-apps install --no-gui {app}'.format(app=app))
+        try:
+            # Install 3rd party apps from Kano World using the App Store.
+            # Before, the disk space requirement was part of the update itself.
+            # Moved away from that model to install apps separately of the update
+            # to minimise the requirement (divide and conquer).
+            # However, the App Store does not check for disk space before
+            # installing an app so account for this here naively.
+            # Requirements (disk_req) in MB were made with apt on a clean system.
+
+            from kano.utils.disk import get_free_space
+
+            new_apps = [
+                {'kw_app': 'tux-paint', 'disk_req': 413},
+                {'kw_app': 'numpty-physics', 'disk_req': 2},
+                {'kw_app': 'gmail', 'disk_req': 1},
+                {'kw_app': 'google-drive', 'disk_req': 1},
+                {'kw_app': 'google-maps', 'disk_req': 1},
+                {'kw_app': 'wikipedia', 'disk_req': 1},
+                {'kw_app': 'whatsapp', 'disk_req': 1},
+                {'kw_app': 'adventure', 'disk_req': 5},
+                {'kw_app': 'openttd', 'disk_req': 21},
+                {'kw_app': 'tux-typing', 'disk_req': 26},
+                {'kw_app': 'libreoffice', 'disk_req': 385},
+            ]
+
+            run_cmd_log('apt-get autoremove -y')
+
+            for app in new_apps:
+                run_cmd_log('apt-get clean')
+
+                mb_free = get_free_space()
+                mb_required = app['disk_req'] + 200  # MB buffer
```

Comment:
Are we happy with 200MB or do we want to use the buffer from the `disk_requirements` module?